Description: Initialize grub_errno before calling get_device_name
Author: Bob Gilligan <gilligan@vyatta.com>
Origin: https://bugzilla.vyatta.com/show_bug.cgi?id=3744

When accessing RAID array with normal mode, grub_file_open would
fail when it should succeed.

The grub_file_open() function uses the global variable grub_errno
as a flag to indicate that an error occurred during the execution
of the grub_file_get_device_name() function.  That function can succeed,
but the error test fail, if grub_errno was set to an error value before
the function call.  The fix is to initialize grub_errno before the
function call.


--- a/kern/file.c	2011-04-26 18:31:28.558824108 -0700
+++ b/kern/file.c	2011-04-26 18:36:18.053849295 -0700
@@ -59,6 +59,7 @@ grub_file_open (const char *name)
   char *device_name;
   char *file_name;
 
+  grub_errno = GRUB_ERR_NONE;
   device_name = grub_file_get_device_name (name);
   if (grub_errno)
     return 0;
